### PR TITLE
Add eini to rebar.lock file

### PR DESCRIPTION
**Pull request by fp**
So if a dependency is not in the lock file and then erlcloud is included as dep by another project, rebar3 will NOT checkout eini! This adds the ref for the tag we are pointing to "fix" this "feature" of rebar3.

### CODE REVIEW

**Comment by fp** [comment]
I have filed an issue with rebar3 about this in: https://github.com/erlang/rebar3/issues/1241

But if we could get this merged quickly it would be greatly appreciated so I can work around the issue.

File: rebar.lock [hunk 1]
```diff
@@ -1,2 +1,6 @@
-[{<<"jsx">>,{pkg,<<"jsx">>,<<"2.8.0">>},0},
+[{<<"eini">>,
```
Comment (motobob) [inline rebar.lock]: @fp may be do like here https://github.com/erlcloud/erlcloud/blob/master/rebar.lock#L1?

Comment (fp) [inline rebar.lock]: I was not aware that `eini` made it into hex yet.  That will certainly work, let me revise.

**Comment by fp** [comment]
Updated to use the `hex.pm` version in the rebar.config as well.

**Pull request merged**